github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml

```diff
@@ -327,19 +327,17 @@ presubmits:
                   - "true"
               topologyKey: kubernetes.io/hostname
       containers:
-      - image: kubevirtci/bootstrap:v20201119-a5880e0
+      - image: quay.io/kubevirtci/golang:v20210316-d295087
+        env:
+          - name: "TARGET"
+            value: "kind-k8s-sriov-1.17.0"
+          - name: "GIMME_GO_VERSION"
+            value: "1.13.8"
         command:
         - "/usr/local/bin/runner.sh"
         - "/bin/bash"
         - "-c"
         - >
-            apt update &&
-            apt install gettext-base -y &&
-            wget https://dl.google.com/go/go1.13.8.linux-amd64.tar.gz -nv -S &&
-            tar -xzf go1.13.8.linux-amd64.tar.gz -C /usr/local/ &&
-            export PATH=/usr/local/go/bin:$PATH &&
-            export KUBEVIRT_PROVIDER=kind-k8s-sriov-1.17.0 &&
-            export TARGET=kind-k8s-sriov-1.17.0;
             trap "echo teardown && make cluster-down" EXIT ERR SIGINT SIGTERM;
```

Comment:
nit: Add && between the trap and automation/test.sh
`trap "echo teardown && make cluster-down" EXIT ERR SIGINT SIGTERM && automation/test.sh`

Reply:
trap can't fail, so i don't see a reason to.
its also not scalable (to add commands) this way, its better to be a command list, than a '&&' list of one command.

Reply:
The problem is that If something goes wrong the job will continue silently.
For example, this is exactly what happen with stiov check cluster-up lane last week
I think you already familiar with #1050, we could prevent the false-positive report if there was && between the trap to other commands.
Not scalable? it dosent prevent one from adding more commands.

Reply:
Trap can't fail.
Its a bad practice to '&&' it to other commands, i am talking about it specifically.
one can add commands below it and '&&' them with themself if needed.

Yes i am familiar with it, and found it also, before knowing about that PR.

We are creating here a bad practice just because with other commands there is a possible to fail.
In case one adds a command, he will need to think if to '&&' it, instead just doing it, just because its there already.

Reply:
What about adding set -e ?
automation/test.sh already have set -e is it won't change the behavior.

```
            set -e;
            trap "echo teardown && make cluster-down" EXIT ERR SIGINT SIGTERM;
            automation/test.sh;
```

/cc @qinqon since you suggested this '&&' first
please advice, and lets wrap it

Reply:
I am ok with `set -e`

Reply:
Please dont add set -e, we already have it on the test.sh script and we put a trap on the command section
that another thing to worry about.

> Trap can't fail.
Its a bad practice to '&&' it to other commands, i am talking about trap specifically.
one can add commands below it and '&&' them with themself if needed.

In case of another bootstrap image change, the shell could also change and bring unexpected behavior.
e.g: trap is not supported, trap syntax change..

Reply:
Sorry, i don't see how trap is going to stop be supported,
and the set -e here is for the command of the job
i find it much better than using && especially when the trap can't fail.
One don't need to worry on this about, on the other hand, if anybody would need to worry about adding && its the not thing to have, this set -e solve it.

Reply:
We have `trap .. ERR` to catch errors, in case more commands should be added to this job it should be added after the trap line. 
Adding set -e is redundant

Reply:
`set -e` cause that a command with an error will exit the "script"
trap ... ERR on its own won't help
hence its not redundant
run this and see done is printed, add set -e and see it doesn't

```
#!/bin/bash

trap "echo teardown" ERR;

false

echo done
```

Reply:
As far as I know "-e" modifier and trap are different things, one make the script fail if one of the commands fail and the other i s use to "arm" the error signal so they complement each other.

I think we can add `set -e` and `trap` so we don't care if the commands are done with `&&` or  `;`

Reply:
> This way we can add lines without infinite '&&'.
Its a bad practice to '&&' commands, especially the trap, which can't fail.
The code should be nice and easy to maintain.

This is not the example I suggested, on your examples replace the ; with && on the trap with and you will see the script will exit.

> As far as I know "-e" modifier and trap are different things, one make the script fail if one of the commands fail and the other i s use to "arm" the error signal so they complement each other.

I think we should test it first

Deleting KIND infra cluster w/o using `kind` binary may cause resources leak (cluster-down use it)
thus its important to test SRIOV lane jobs before merging.
After we finish with this job we can continue with release branches jobs.
Please inject an error so we could see that it does fail and cluster-down is invoked once.

Reply:
I tested it here
https://github.com/kubevirt/project-infra/pull/1068
it runs the trap.